pkg/file/builder.go

```diff
@@ -967,6 +973,29 @@ func (b *stateBuilder) vaults() {
 	}
 }
 
+func (b *stateBuilder) licenses() {
+	if b.err != nil {
+		return
+	}
+
+	for _, l := range b.targetContent.Licenses {
+		l := l
+		if utils.Empty(l.ID) {
+			license, err := b.currentState.Licenses.Get(*l.ID)
```

Comment:
If `l.ID` is empty we use it in `Get()`? I'm not sure I understand the purpose here 🤔 Might be worth adding a comment about this and/or changing this.

Reply:
Fixed. This was copied from other types but `license` only have one identifier `ID`, so I think the proper way to process when ID is not given is to generate an ID.